tests/capi/unittest_capi_service_extension.cc

```diff
@@ -0,0 +1,1250 @@
+
+  status = ml_service_new (config, &handle);
+  ASSERT_EQ (status, ML_ERROR_NONE);
+
+  status = ml_service_set_information (handle, NULL, "0.1");
+  EXPECT_NE (status, ML_ERROR_NONE);
+  status = ml_service_set_information (handle, "", "0.1");
+  EXPECT_NE (status, ML_ERROR_NONE);
+  status = ml_service_set_information (handle, "test-threshold", NULL);
+  EXPECT_NE (status, ML_ERROR_NONE);
+  status = ml_service_set_information (handle, "test-threshold", "");
+  EXPECT_NE (status, ML_ERROR_NONE);
+
+  status = ml_service_destroy (handle);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Testcase with invalid param.
+ */
+TEST (MLServiceExtension, getInfoInvalidParam01_n)
+{
+  char *value;
+  int status;
+
+  status = ml_service_get_information (NULL, "threshold", &value);
+  EXPECT_NE (status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Testcase with invalid param.
+ */
+TEST_REQUIRE_TFLITE (MLServiceExtension, getInfoInvalidParam02_n)
+{
+  ml_service_h handle;
+  char *value;
+  int status;
+
+  g_autofree gchar *config = _get_config_path ("config_single_imgclf.conf");
+
+  status = ml_service_new (config, &handle);
+  ASSERT_EQ (status, ML_ERROR_NONE);
+
+  /* Set invalid magic. */
+  TEST_SET_MAGIC (handle, 0U);
+  status = ml_service_get_information (handle, "threshold", &value);
+  EXPECT_NE (status, ML_ERROR_NONE);
+  TEST_SET_MAGIC (handle, 0xfeeedeed);
+
+  status = ml_service_get_information (handle, "threshold", &value);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+  EXPECT_STREQ (value, "0.5");
+  g_free (value);
+
+  status = ml_service_destroy (handle);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Testcase with invalid param.
+ */
+TEST_REQUIRE_TFLITE (MLServiceExtension, getInfoInvalidParam03_n)
+{
+  ml_service_h handle;
+  char *value;
+  int status;
+
+  g_autofree gchar *config = _get_config_path ("config_single_imgclf.conf");
+
+  status = ml_service_new (config, &handle);
+  ASSERT_EQ (status, ML_ERROR_NONE);
+
+  status = ml_service_get_information (handle, NULL, &value);
+  EXPECT_NE (status, ML_ERROR_NONE);
+  status = ml_service_get_information (handle, "", &value);
+  EXPECT_NE (status, ML_ERROR_NONE);
+  status = ml_service_get_information (handle, "invalid_name", &value);
+  EXPECT_NE (status, ML_ERROR_NONE);
+
+  status = ml_service_destroy (handle);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Testcase with invalid param.
+ */
+TEST_REQUIRE_TFLITE (MLServiceExtension, getInfoInvalidParam04_n)
+{
+  ml_service_h handle;
+  int status;
+
+  g_autofree gchar *config = _get_config_path ("config_single_imgclf.conf");
+
+  status = ml_service_new (config, &handle);
+  ASSERT_EQ (status, ML_ERROR_NONE);
+
+  status = ml_service_get_information (handle, "threshold", NULL);
+  EXPECT_NE (status, ML_ERROR_NONE);
+
+  status = ml_service_destroy (handle);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Testcase with invalid param.
+ */
+TEST (MLServiceExtension, addInputInvalidParam01_n)
```

Comment:
Let's change the test name
```suggestion
TEST (MLServiceExtension, requestInvalidParam01_n)
```

Reply:
thanks, updated.